packages/core/src/lib/modal.svelte

```diff
@@ -26,31 +26,41 @@ Creates a modal overlay.
 
 <script lang="ts">
 import cx from 'classnames';
-import { createEventDispatcher, onMount } from 'svelte';
+import { createEventDispatcher, onMount, afterUpdate } from 'svelte';
 import IconButton from './button/icon-button.svelte';
+import { clickOutside } from '$lib';
 
 /** Whether the modal is open. */
 export let open = false;
 /** The variant of the modal. */
 export let variant: 'small' | '' = '';
 
+let clickOutsideActive = false;
+
 const dispatch = createEventDispatcher<{
   /** Fires when the modal is requested to close. */
   close: boolean;
```

Comment:
Does this event really need a payload?

Reply:
Thanks for the suggestion!  Updated to accept an isOpen writable store